### PR TITLE
feature(devtools): migrate from local repo to floating ui repo

### PR DESCRIPTION
**Pull request by bsunderhus**
Proposal: migrate devtools from personal repo (https://github.com/bsunderhus/floating-ui-devtools) to floating ui repo

Follow up of: https://github.com/floating-ui/floating-ui/issues/2649

### CODE REVIEW

File: extension/.storybook/decorators.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import { FluentProvider, webDarkTheme, webLightTheme } from '@fluentui/react-components';
```
Comment (FezVrasta) [inline extension/.storybook/decorators.tsx:2]: Could you help me understand why we want to tie this to Fluent UI?

Comment (bsunderhus) [reply extension/.storybook/decorators.tsx:2]: We don't, I'll move it to the Fluent example, thx for the catch!

**Comment by bsunderhus** [comment]
I'll keep this as draft for now as I still need to clean up some things

1. add proper readmes and contribution guides
2. properly integrate with the way things work in floating ui monorepo

**Comment by bsunderhus** [comment]
Sorry upfront for the huge PR! Here's a sum up:

1. creates an `extension` folder containing a private package to build the chrome extension
2. creates a `packages/devtools` folder containinig everything related to the devtools middleware package exposing the minimal amount of information.
3. update tsconfig files to exclude `packages/devtools` from it's type verifications, as that package verification is not compatible
4. add estlint rule to avoid the usage of `chrome` global namespace

The extension is shipping by default with support to raw usages of floating ui and also supporting the FluentUI library use case. The idea here is that component libraries (such as fluent) will be using floating ui internally, and they might be interested in adding their own custom panel to ensure their clients have the support they need.

I've added a session in the REAMDE file of the extension folder explaining how you can add your own custom library middleware data (would be nice to have other major libraries implementing their own custom panel here, like Material UI)


## Next Steps

1. The `SidePanel` used to add extra helper functionalities is currently implemented using Fluent, this leaks FluentUI to the main bundle, ideally we should use a custom UI there instead of FluentUI
2. rename `floating-ui-devtools` to `@floating-ui/devtools` to follow other packages
3. follow up on https://github.com/floating-ui/floating-ui/issues/2649 todos

**Comment by bsunderhus** [comment]
> Thanks this looks great! Any plan to add a CI step to publish the extension as well?

Yeah, that would be the next step, together with renaming the package from `floating-ui-devtools` to `@floating-ui/devtools` I can do this in this PR also, I'll update later on today

**Comment by bsunderhus** [comment]
Ok, I've added the building process of `@floating-ui/devtools` to the `build` script which will be executed by the `release` CI if I understood correctly @FezVrasta, also change the name of the package and I guess this will trigger a release as this package has never been published before. Is this enough to include the package in the release pipeline?

**Comment by FezVrasta** [comment]
I'll need @atomiks review here as I'm not informed on how the floating-ui new repo architecture works but what I see looks solid to me.

Regarding the build process, I hoped we could have a GitHub Actions workflow that directly publishes to the store any new version. Where I work we use `chrome-webstore-upload-cli` to make it happen

**Comment by bsunderhus** [comment]
> I'll need @atomiks review here as I'm not informed on how the floating-ui new repo architecture works but what I see looks solid to me.
> 
> Regarding the build process, I hoped we could have a GitHub Actions workflow that directly publishes to the store any new version. Where I work we use `chrome-webstore-upload-cli` to make it happen

I'll need to register tokens on the Github Actions Pipeline for that, and I can't do that at the moment as I'm not an internal contributor to the repository with this permissions. How about we leave this automation for a separate PR?

**Comment by atomiks** [comment]
Thank you for this PR! I'm sorry I didn't get to look at it until now. Taking a look now...

**Comment by atomiks** [comment]
This is a big PR as you mentioned, but from briefly testing the extension and the local devtools package it appears to work as expected. It's probably going to be hard for me to personally contribute to the extension, but as long as you have the ownership of it (PRs etc) then it seems fine to add to this repo. 

Thank you!

**Comment by FezVrasta** [comment]
Are we ready to merge or there's some work left? Just let me know and I'll hit the button

**Comment by bsunderhus** [comment]
Thx for the review @atomiks! I'll be following up with the contribution to maintain the extension for sure. And let me know if we can create a token in the pipeline to automate the publishing of the extension, in the meantime I'll keep the publications manually.

This is ready to be merged @FezVrasta , once it's merged I'll update https://github.com/floating-ui/floating-ui/issues/2649 with next steps!

**Comment by FezVrasta** [comment]
Thanks, if you want to send me the token by email you can reach me at federico.zivolo (AT) gmail

**Comment by karthik15981** [comment]
image

This change is breaking in Web Pack 4..

**Comment by bsunderhus** [comment]
> image
> 
> This change is breaking in Web Pack 4..

@karthik15981 I'd guess the problem here is that your webpack configuration does not support [Nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) operator?!


@FezVrasta @atomiks , what is the ES target for floating-ui, by the configuration files I'd guess it's ES2020, no?! should I strip away nullish coalescing operators from the distribution?

**Comment by karthik15981** [comment]
@bsunderhus Yes, it's the nullish coalescing operator that's breaking in Web Pack 4..

**Comment by atomiks** [comment]
The other packages do transpile the nullish coalescing already; I think the `devtools` package needs a `babel.config.js` file for Rollup to read from, then it should transpile it. (Need to verify if that's actually what causes it to be transpiled tho)

**Comment by bsunderhus** [comment]
The devtools package is bundling with vite actually. I can create a PR to migrate it to the same rollup configuration as the other packages for the sake of following a pattern

**Comment by atomiks** [comment]
Another thing is, the dist files should be `.js` for cjs and fake esm, and `.mjs` for node esm, since `.cjs` isn't supported in Create React App or something...

> The devtools package is bundling with vite actually. I can create a PR to migrate it to the same rollup configuration as the other packages for the sake of following a pattern

Yeah, that would also solve the above issue.

**Pull request merged**